src/vivarium_gates_nutrition_optimization_child/components/observers.py

```diff
@@ -9,77 +9,96 @@
-#         # applicable to stunting and wasting
-#         return {
-#             "cat4": data_keys.CGFCategories.UNEXPOSED.value,
-#             "cat3": data_keys.CGFCategories.MILD.value,
-#             "cat2": data_keys.CGFCategories.MODERATE.value,
-#             "cat1": data_keys.CGFCategories.SEVERE.value,
-#         }[row.squeeze()]
+from vivarium_gates_nutrition_optimization_child.constants import data_keys, results
+
+
+class ResultsStratifier(ResultsStratifier_):
+    """Centralized component for handling results stratification.
+    This should be used as a sub-component for observers.  The observers
+    can then ask this component for population subgroups and labels during
+    results production and have this component manage adjustments to the
+    final column labels for the subgroups.
+    """
+
+    def register_stratifications(self, builder: Builder) -> None:
+        """Register each desired stratification with calls to _setup_stratification"""
+        super().register_stratifications(builder)
+
+        builder.results.register_stratification(
+            "wasting_state",
+            [category.value for category in data_keys.CGFCategories],
+            self.child_growth_risk_factor_stratification_mapper,
+            is_vectorized=True,
+            requires_values=["child_wasting.exposure"],
+        )
+        builder.results.register_stratification(
+            "stunting_state",
+            [category.value for category in data_keys.CGFCategories],
+            self.child_growth_risk_factor_stratification_mapper,
+            is_vectorized=True,
+            requires_values=["child_stunting.exposure"],
+        )
+        builder.results.register_stratification(
+            "maternal_supplementation",
+            results.MATERNAL_SUPPLEMENTATION_TYPES,
+            is_vectorized=True,
+            requires_columns=["maternal_supplementation_exposure"],
+        )
+        builder.results.register_stratification(
+            "iv_iron",
+            results.DICHOTOMOUS_COVERAGE_STATES,
+            is_vectorized=True,
+            requires_columns=["iv_iron_exposure"],
+        )
+        builder.results.register_stratification(
+            "bmi_anemia",
+            ["cat4", "cat3", "cat2", "cat1"],
+            is_vectorized=True,
+            requires_columns=["maternal_bmi_anemia_exposure"],
+        )
+
+    ###########################
+    # Stratifications Details #
+    ###########################
+
+    # noinspection PyMethodMayBeStatic
+    def child_growth_risk_factor_stratification_mapper(self, pop: pd.DataFrame) -> pd.Series:
+        # applicable to stunting and wasting
+        mapper = {
+            "cat4": data_keys.CGFCategories.UNEXPOSED.value,
+            "cat3": data_keys.CGFCategories.MILD.value,
+            "cat2": data_keys.CGFCategories.MODERATE.value,
+            "cat1": data_keys.CGFCategories.SEVERE.value,
+        }
+        return pop.squeeze(axis=1).replace(mapper)
```

Comment:
Note: If you use `replace` values not in the dictionary's keys will not be changed, whereas if you use `map` they will be converted to NaN. It should make no difference in this case, since if the mapper outputs anything that is not in the accepted list of categories it will error out, but in general, I think `map` seems safer unless you actually don't want to change the elements that don't appear in the dictionary.